performance_manager/lib/logging_utils.py

```diff
@@ -18,70 +17,75 @@ def __init__(self, process_name: str, **metadata: MdValues) -> None:
         create a process logger with a name and optional metadata. a start time
         and uuid will be created for timing and unique identification
         """
-        self.parent = "performance_manager"
-        self.process_name = process_name
-        self.metadata = metadata
-        self.uuid: Optional[uuid.UUID] = None
+        self.default_data: Dict[str, Any] = {}
+        self.metadata: Dict[str, Any] = {}
+
+        self.default_data["parent"] = "performance_manager"
+        self.default_data["process_name"] = process_name
+
+        # default_data keys that can not be added as metadata
+        self.protected_keys = [
+            "parent",
+            "process_name",
+            "process_id",
+            "uuid",
+            "status",
+            "duration",
+            "error_type",
+        ]
+
         self.start_time = 0.0
-        self.process_id = 0
+
+        self.add_metadata(**metadata)
+
+    def _get_log_string(self) -> str:
+        """create logging string for log write"""
+        logging_list = []
+        # add default data to log output
+        for (key, value) in self.default_data.items():
+            logging_list.append(f"{key}={value}")
+
+        # add metadata to log output
+        for (key, value) in self.metadata.items():
+            logging_list.append(f"{key}={value}")
+
+        return ", ".join(logging_list)
```

Comment:
unified log string creator. log items will be added to log string in order they were added to dictionary objects. all `default_data` logs items will appear before `metadata` log items